pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,189 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+
+	"kubevirt.io/client-go/kubecli"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+)
+
+func GetNetworkToIPAMClaimName(virtClient kubecli.KubevirtClient, namespace string, name string, networks []virtv1.Network) (map[string]string, error) {
```

Comment:
There are a lot of stuff going on here and also it's plural since we return multiple IPAMClaims name.
This should be done at two steps (not doing it for performance reason is an early optimization)

1.- Filter networks by IsSecondaryMultus and allowPersistentIps at a separate function using `vmispec.FilterNetworksSpec`
2.- With the output from function 1 compose IPAMClaim names at a different function.

If we forget about early optimization we should use `vmispec.FilterNetworksByInterfaces(networks, nonAbsentIfaces)